tools/mock-service-host/CHANGELOG.md

```diff
@@ -1,5 +1,14 @@
 # Change Log - @azure-tools/mock-service-host
 
+## 0.1.16
+2022-05-27
+
+### Patches
+
+- Use example response prior to mock response
+- Use 200 response if can't find LRO callback url
+- Fix resource-type mocking for list operation 
```

Comment:
Don't quite understand 'Fix resource-type mocking for list operation '  could you explain more about this?

Reply:
Hi @lirenhe , this is a case of setting resource-id/resource-type for those list operations that don't have same path pattern with resource creation/reading path.

For example: 
For list operation https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json#L1024,
the element resource type of the response is https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json#L2146,
and the get instance operation for this resource type is at https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json#L1063,
So though the list operation path is "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/deletedConfigurationStores", the resources it return is of path "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/locations/{location}/deletedConfigurationStores/{configStoreName}".
I feel we should do this logic because the live recording and swagger example of this list operation have this behavior: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/examples/DeletedConfigurationStoresList.json#L11

Reply:
You mean the example's resource id is not correct. Right?

Reply:
The resource id in example is correct according to the live test recording in .net.

In this case, the list operation "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/deletedConfigurationStores" actually return the instances of "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/locations/{location}/deletedConfigurationStores/xxx"

Originally the mocked resource id is "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/deletedConfigurationStores/xxx", this PR is trying to handle the above situation.

Reply:
As this is a special case, will it cause problem if we return the original expected id?  like 
/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/deletedConfigurationStores/xxx

I think in the Swagger definition, the type of id is just a string.

Reply:
Yes it's a special case, and most SDK is fine with current situation, but .Net SDK reports error on "/subscriptions/{subscriptionId}/providers/Microsoft.AppConfiguration/deletedConfigurationStores/xxx" since it validates the returned resource type.

Reply:
One thinking on this,  instead of introducing this complex logic, is it possible to put it as the directive or configuration for mock service host so we won't take it for granted and this is likely to be an issue we need to catch?

Open for discussion, cc @tadelesh  @ArcturusZhang

Reply:
I agree with the directive implementation. The swagger examples will still contain different kinds of unexpected issues in a period of time even if the s360 pipeline is on the way as Ray told me. It will be much more flexible if we have some configuration to change the example temporally.

Reply:
for the implementation, I think we can do configuration for this by similar way in current .env? The mock-service-host don't understand autorest like directives

Reply:
This behaviour is a violation against the existing ones so I doubt whether mock-service-host should run correctly by default.
I am ok with using .env to store config but you and @tadelesh could discuss what the best place is.